app/domain/authentication/authn_gcp/authenticator.rb

```diff
@@ -3,29 +3,47 @@
 module Authentication
   module AuthnGcp
 
+    # This authenticator validates that the received token matches the
+    # restrictions defined in the policy. There is another validation step
+    # where we also verify that the token is valid (JWT-wise) but this is
+    # done while it is decoded (using a third-party).
     Authenticator = CommandClass.new(
       dependencies: {
-        validate_resource_restrictions: ValidateResourceRestrictions.new,
+        validate_resource_restrictions: Authentication::ResourceRestrictions::ValidateResourceRestrictions.new,
```

Comment:
here i see that you have a `ResourceRestrictions` module. it can be added to the error classes as well.